sk0-front/src/app/guards/auth.guard.ts

```diff
@@ -0,0 +1,12 @@
+import { CanActivateFn, Router } from '@angular/router';
+import { AuthService } from '../auth/auth.service';
+import { inject } from '@angular/core';
+
+export const authGuard: CanActivateFn = () => {
+  const router = inject(Router);
+  const authService = inject(AuthService);
+  if (authService.getIsUserLoggedIn()) {
+    return true;
+  }
+  return router.navigate(['login']);
```

Comment:
Unfortunately this variable is not reactive. It should be an Observable or Promise.

Reply:
if the `isUserLoggedIn$` is a `Subject` then it *might* be only 
```ts
export const authGuard: CanActivateFn = () => inject(AuthService).isUserLoggedIn$
```

Reply:
Thats fair. I saw that this variable is changed to Behaviorsubject in the authService login and register functions PR. Should I wait with the correction until that branch is merged into dev, or should I cherry-pick those commits into this branch?

Reply:
Well I would merge it like this. My commits are most of the times rather guidance than requirements. We should fix it when we migrate to reactive state management (subjects/observables).